src/macros.rs

```diff
@@ -54,68 +384,104 @@ bitvec![1; 5];
-		bitvec![ __bv_impl__ $order , $crate::store::Word ; $( $val ),* ]
+	($order:ident; $($val:expr),* $(,)?) => {
+		$crate::vec::BitVec::<
+			$order,
+			$crate::store::Word,
+		>::from_bitslice($crate::__bits_from_slice_word!($order; $($val),*))
+	};
+	($order:path; $($val:expr),* $(,)?) => {
+		$crate::vec::BitVec::<
+			$order,
+			$crate::store::Word,
+		>::from_bitslice($crate::__bits_from_slice_word!($order; $($val),*))
 	};
-	//  bitvec![ endian ; 0 , 1 , … , ]
-	( $order:path ; $( $val:expr , )* ) => {
-		bitvec![ __bv_impl__ $order , $crate::store::Word ; $( $val ),* ]
+	($($val:expr),* $(,)?) => {
+		$crate::vec::BitVec::<
+			$crate::order::Local,
+			$crate::store::Word,
+		>::from_bitslice($crate::__bits_from_slice_word!(Local; $($val),*))
 	};
 
-	//  bitvec![ 0 , 1 , … ]
-	( $( $val:expr ),* ) => {
-		bitvec![ __bv_impl__ $crate::order::Local , $crate::store::Word ; $( $val ),* ]
+	($order:ident, $bits:ident; $val:expr; $len:expr) => {
+		$crate::vec::BitVec::<$order, $bits>::repeat($val != 0, $len)
 	};
-	//  bitvec![ 0 , 1 , … , ]
-	( $( $val:expr , )* ) => {
-		bitvec![ __bv_impl__ $crate::order::Local , $crate::store::Word ; $( $val ),* ]
+	($order:path, $bits:ident; $val:expr; $len:expr) => {
+		$crate::vec::BitVec::<$order, $bits>::repeat($val != 0, $len)
 	};
-
-	//  bitvec![ endian , type ; bit ; rep ]
-	( $order:path , $bits:ty ; $val:expr ; $rep:expr ) => {
-		bitvec![ __bv_impl__ $order , $bits ; $val; $rep ]
+	($order:ident; $val:expr; $len:expr) => {
+		$crate::vec::BitVec::<
+			$order,
+			$crate::store::Word,
+		>::repeat($val != 0, $len)
 	};
-	//  bitvec![ endian ; bit ; rep ]
-	( $order:path ; $val:expr ; $rep:expr ) => {
-		bitvec![ __bv_impl__ $order , $crate::store::Word ; $val ; $rep ]
+	($order:path; $val:expr; $len:expr) => {
+		$crate::vec::BitVec::<
+			$order,
+			$crate::store::Word,
+		>::repeat($val != 0, $len)
 	};
-	//  bitvec![ bit ; rep ]
-	( $val:expr ; $rep:expr ) => {
-		bitvec![ __bv_impl__ $crate::order::Local , $crate::store::Word ; $val ; $rep ]
+	($val:expr; $len:expr) => {
+		$crate::vec::BitVec::<
+			$crate::order::Local,
+			$crate::store::Word,
+		>::repeat($val != 0, $len)
 	};
+}
 
-	//  GitHub issue #25 is to make this into a proc-macro that produces the
-	//  correct memory slab at compile time.
+// Not public API
+#[cfg(feature = "alloc")]
+#[doc(hidden)]
+#[macro_export]
+macro_rules! __bitvec_from_slice {
+	// Ensure literal `Msb0`, `Lsb0`, and `Local` always map to our internal
+	// definitions of the type, as we've been structurally matching them.
+	(Lsb0, $bits:ident, $slice:expr) => {
+		$crate::vec::BitVec::<
+			$crate::order::Lsb0,
+			$bits,
+		>::from_slice(&$slice[..])
+	};
+	(Msb0, $bits:ident, $slice:expr) => {
+		$crate::vec::BitVec::<
+			$crate::order::Msb0,
+			$bits,
+		>::from_slice(&$slice[..])
+	};
+	(Local, $bits:ident, $slice:expr) => {
+		$crate::vec::BitVec::<
+			$crate::order::Local,
+			$bits,
+		>::from_slice(&$slice[..])
+	};
```

Comment:
It'd also be fine to make `bits` take a `Path` here. We only need it to be an `ident` for the interface, because only `Local`, `u8`, `u16`, `u32`, and `u64` are supported.